pages/upgrade-guide/4.0.md

```diff
@@ -0,0 +1,186 @@
+
+jQuery 3.0.0 had already stopped auto-executing scripts retrieved with `jQuery.ajax` unless `dataType: "script"` was explicitly set for cross-domain scripts. That change has now been extended to same-domain scripts.
+
+To fetch *and* evaluate a script, pass `dataType: "script"` in `jQuery.ajax` options or use `jQuery.getScript`.
+
+#### Breaking change: Script tags now used for all async requests
+
+Until jQuery 4.0, the AJAX script transport only used a script tag to load scripts for cross-domain requests or ones with `scriptAttrs` set. We now always use a script tag for async requests to avoid any [Content Security Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) errors.
+
+This change may affect existing code if certain scripts were expected to execute even when CSP headers were set. Either modify the CSP header to allow scripts from the same domain or remove the scripts.
+
+### Attributes
+
+#### Breaking change: `toggleClass(boolean | undefined)` signature removed
+
+This particular signature was deprecated in jQuery 3.0 and is now being removed in jQuery 4.0. After feedback and discussion with the community, it was determined that this signature caused unexpected behavior. For instance, if the argument to `toggleClass` was a variable that turned out to be `undefined` or `false`, it would result in all classes being removed. Also, while the `toggleClass( boolean )` signature was documented (and has been marked deprecated since 3.0), the `toggleClass( undefined )` signature was not.
+
+The recommendation for migration is to be explicit about which classes should be toggled.
+
+### Core
+
+#### Breaking change: Deprecated APIs removed
+
+The following APIs were deprecated in previous versions and have been removed in jQuery 4.0:
+
+* `jQuery.camelCase`
+	This function was never documented and was only used internally. It was deprecated in jQuery 3.0.
+
+* `jQuery.cssProps`
+	This object was used to map CSS properties to their vendor-prefixed versions, which is no longer necessary in jQuery 4.0's supported browsers.
+
+* `jQuery.isArray`
+* `jQuery.isFunction`
+* `jQuery.isNumeric`
+* `jQuery.isWindow`
+* `jQuery.fx.interval`
+* `jQuery.parseJSON`
+* `jQuery.nodeName`
+* `jQuery.now`
+* `jQuery.trim`
+* `jQuery.type`
+
+#### Breaking change: Remove undocumented Array methods
+
+jQuery 4.0 removes `push`, `sort`, and `splice` from the jQuery prototype. These methods were copies of the methods of the same names on the native Array prototype, but were never documented and were not intended for public use. Some plugins may have used them, so we are still marking this as a breaking change.
+
+#### Slim build breaking change: Remove `callbacks` and `deferred` modules
+
+The slim build of jQuery 4.0 no longer includes the `callbacks` and `deferred` modules. Use the full build if you need these modules, or use native Promise objects.
+
+#### Breaking change: Remove undocumented `root` parameter of `jQuery()`
```

Comment:
Not of `jQuery()`; of `jQuery.fn.init()`. `jQuery` is already skipping this argument even on `3.x-stable` (https://github.com/jquery/jquery/blob/3.x-stable/src/core.js#L33-L38):
```js
	jQuery = function( selector, context ) {

		// The jQuery object is actually just the init constructor 'enhanced'
		// Need init if jQuery is called (just allow error to be thrown if not included)
		return new jQuery.fn.init( selector, context );
	};
```
while in `jQuery.fn.init` it exists there (https://github.com/jquery/jquery/blob/3.x-stable/src/core/init.js):
```js
	init = jQuery.fn.init = function( selector, context, root ) {
```

Let's also mentioned this is the third parameter.